### PR TITLE
Add cargo deny config and add cargo deny job to ci

### PR DESCRIPTION
**Pull request by iamjpotts**
Add `cargo deny` configuration which matches status quo of the crate. Also add a job to CI to run the check.

This will cause CVEs, duplicate dependencies, unexpected licenses, and other issues to be caught during CI, before the crate is published.

### CODE REVIEW

**Review by Stebalien — APPROVED** [review]
Thanks!

**Comment by Stebalien** [comment]
Hm. So, the job is _really_ slow because we have to compile it. Could you switch to downloading a pre-compiled binary from https://github.com/EmbarkStudios/cargo-deny/releases/tag/0.14.1?

**Comment by iamjpotts** [comment]
> Hm. So, the job is _really_ slow because we have to compile it. Could you switch to downloading a pre-compiled binary from https://github.com/EmbarkStudios/cargo-deny/releases/tag/0.14.1?

There's a [github action](https://github.com/EmbarkStudios/cargo-deny-action) referenced on that page, I'm going to try that first. The first commit with have two copies of the job, so I can inspect and compare their results and timings.

**Comment by iamjpotts** [comment]
It looks like the CI job is awaiting repo maintainer approval to run.

File: .gitignore
Comment (Stebalien) [inline .gitignore:7]: Please add these to your global gitignore instead (e.g., https://gist.github.com/subfuzion/db7f57fff2fb6998a16c). Otherwise, we end up with an entry for every editor/ide in every project.

**Comment by Stebalien** [comment]
LGTM modulo the gitignore changes.

**Comment by iamjpotts** [comment]
Of the two flavors of github actions for cargo deny I added, which would you prefer to keep?

A) The flavor that uses a specific version of Rust, which we can maintain to match this crate's minimum rust version, and takes 3 minutes to run
B) The flavor that uses whatever version of Rust is packaged with the Embark action, may be newer or older than this crate's minimum rust version, but only takes 30 seconds to run

A, or B? B, or A? (as they eye doctor would say)

**Comment by Stebalien** [comment]
Ah, I see. Keep the precompiled version. As far as I know, the rust version shouldn't matter much unless I start using a new cargo feature, at which point I can upgrade it.

**Comment by iamjpotts** [comment]
Got rid of the gitignore change, and consolidated the deny jobs to just the one using the pre-built binaries.

Workflow is pending approval again (probably due to it being edited).

**Comment by iamjpotts** [comment]
> Ah, I see.

Pardon the pun :)

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
Thanks! And I swear the pun was unintended :)

**Pull request merged**
